realsense2_camera/src/base_realsense_node.cpp

```diff
@@ -150,18 +150,19 @@ BaseRealSenseNode::~BaseRealSenseNode()
     std::set<std::string> module_names;
     for (const std::pair<stream_index_pair, std::vector<rs2::stream_profile>>& profile : _enabled_profiles)
     {
-        std::string module_name = _sensors[profile.first].get_info(RS2_CAMERA_INFO_NAME);
-        std::pair< std::set<std::string>::iterator, bool> res = module_names.insert(module_name);
-        if (res.second)
+        try
```

Comment:
Is this change necessary? Did you encounter an exception in the area not previously inside the try-catch section?

Reply:
At one point yes, we were seeing some unhandled realsense exceptions in the destructor.  This was originally developed against an older version of librealsense though, so it's possible those have been fixed in the latest version.  So to some degree, this may be preventative now.